### PR TITLE
Restructure configuration schema and deprecate legacy config keys

### PR DESCRIPTION
**Pull request by ssheng**
## Description

Restructured configuration schema in `bentoml.yml` to a more logical and hierarchical format.
```
bento_server:
  port: 5000
  ngrok:
    enabled: False
  workers: 1
  timeout: 60
  max_request_size: 20971520
  microbatch:
    enabled: False
    workers: 1
    max_batch_size: Null
    max_latency: Null
  swagger:
    enabled: True
  metrics:
    enabled: True
    namespace: BENTOML
  feedback:
    enabled: True
  logging:
    level: INFO

tracing:
  type: zipkin
  zipkin:
    url: Null
  jaeger:
    address: Null
    port: Null

adapters:
  image_input:
    extensions:
      - .jpg
      - .png
      - .jpeg
      - .tiff
      - .webp
      - .bmp
```

Deprecated the legacy `core.debug` key from the current configuration `bentoml.cfg`. Replaced debug capability with an environment variable `BENTOML_DEBUG` option. The CLI parameters `--verbose` and `--debug` will still work as before.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature and improvements (non-breaking change which adds/improves functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Code Refactoring (internal change which is not user facing)
- [ ] Documentation
- [ ] Test, CI, or build

## Component(s) if applicable
- [ ] BentoService (service definition, dependency management, API input/output adapters)
- [ ] Model Artifact (model serialization, multi-framework support)
- [ ] Model Server (mico-batching, dockerisation, logging, OpenAPI, instruments)
- [ ] YataiService gRPC server (model registry, cloud deployment automation)
- [ ] YataiService web server (nodejs HTTP server and web UI)
- [ ] Internal (BentoML's own configuration, logging, utility, exception handling)
- [ ] BentoML CLI

## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/configuration/default_bentoml.yml [hunk 1]
```diff
@@ -43,8 +33,12 @@ tracing:
...
     port: Null
 
-instrument:
-  namespace: BENTOML
-
-logging:
-  level: INFO
+adapters:
+  image_input:
+    extensions:
```
Comment (ssheng) [inline bentoml/configuration/default_bentoml.yml:38]: Rename to `default_extensions`.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -32,21 +32,13 @@ To give us more information for diagnosing the issue, make sure to enable debug
...
 ```
 $ git clone git@github.com:bentoml/BentoML.git && cd bentoml
-$ BENTOML__CORE__DEBUG=true python guides/quick-start/main.py
-```
-
-Or enable for all python sessions on current machine:
-```bash
-$ bentoml config set core.debug=true
-$ python guides/quick-start/main.py
+$ BENTOML_DEBUG=TRUE python guides/quick-start/main.py
```
Comment (parano) [inline .github/ISSUE_TEMPLATE/bug_report.md:35]: 👍

File: tests/configuration/test_configuration.py [hunk 3]
```diff
@@ -13,3 +20,71 @@ def test_get_local_config_file():
...
+
+
+def assert_debug_mode(enabled: bool):
+    if get_debug_mode() is enabled:
+        sys.exit(0)
+    else:
+        sys.exit(1)
+
+
+def test_multiprocess_debug_mode():
```
Comment (parano) [inline tests/configuration/test_configuration.py:65]: 👍

**Comment by parano** [comment]
Looks great, thanks @ssheng!!

**Pull request merged**